app/controllers/candidate_interface/gcse_details_controller.rb

```diff
@@ -1,15 +1,76 @@
 module CandidateInterface
   class GcseDetailsController < CandidateInterfaceController
-    def edit
-      subject = params[:subject]
+    before_action do
+      @subject = subject_param
+    end
+
+    # 1th step - Edit qualification type
+    def edit_type
+      @application_qualification = GcseQualificationTypeForm.new(
+        subject: subject_param,
+        level: ApplicationQualification.levels[:gcse],
+      )
+    end
+
+    def update_type
+      @application_qualification = GcseQualificationTypeForm.new(qualification_type: qualification_type_param,
+                                                                 subject: subject_param,
+                                                                 level: ApplicationQualification.levels[:gcse])
+
+      application_form = current_candidate.current_application
+
+      if @application_qualification.save_base(application_form)
+        redirect_to candidate_interface_gcse_details_edit_details_path
+      else
+        render :edit_type
+      end
+    end
+
+    # 2nd step - Edit grade and award year
+    def edit_details
+      application_qualification = ApplicationQualification.last
+
+      @application_qualification = GcseQualificationDetailsForm.new(
+        grade: application_qualification.grade,
+        award_year: application_qualification.award_year,
+      )
+
+      render :edit_details
+    end
+
+    def update_details
+      current_application_qualification = ApplicationQualification.last
+
+      details_form = GcseQualificationDetailsForm.new(
+        grade: params[:candidate_interface_gcse_qualification_details_form][:grade],
+        award_year: params[:candidate_interface_gcse_qualification_details_form][:award_year],
+      )
+
+      @application_qualification = details_form.save_base(current_application_qualification)
+
+      if @application_qualification
+        redirect_to candidate_interface_gcse_review_path
+      else
+        @application_qualification = details_form
 
-      @application_qualification = ApplicationQualification.create(subject: subject, application_form: current_candidate.current_application, level: 'gcse')
+        render :edit_details
+      end
     end
 
-    def update
+    def review
       @application_qualification = ApplicationQualification.last
 
-      render :show
+      render :review
+    end
+
+  private
+
+    def subject_param
+      params.require(:subject)
+    end
+
+    def qualification_type_param
+      (params[:candidate_interface_gcse_qualification_type_form] || {}).fetch(:qualification_type, '')
     end
```

Comment:
Would doing something like this but for GCSEs achieve the same thing?

```ruby
def degrees_params
  params.require(:candidate_interface_degrees_form).permit(
    :qualification_type, :subject, :institution_name, :grade, :other_grade,
    :predicted_grade, :award_year
  )
end
```

Reply:
that does not work because subject is not part of the form (`candidate_interface_gcse_qualification_type_form`), it is part of the URL param,
I'll see if I can make some improvements on the following PR